rest-openehr/src/main/java/org/ehrbase/rest/openehr/OpenehrEhrController.java

```diff
@@ -77,213 +67,116 @@ public OpenehrEhrController(EhrService ehrService, SystemService systemService)
-        // check for valid format header to produce content accordingly
-        MediaType contentType = resolveContentType(accept);
+    private ResponseEntity.BodyBuilder ehrResponseBuilder(HttpStatus status, UUID ehrId) {
 
-        // create either null or maximum response data class
-        T minimalOrRepresentation = factory.get();
+        URI uri = createLocationUri(EHR, ehrId.toString());
 
-        if (minimalOrRepresentation != null) {
-
-            EhrStatus ehrStatus = ehrService.getEhrStatus(ehrId);
-            // populate maximum response data
-            EhrResponseData objByReference = minimalOrRepresentation;
-            objByReference.setEhrId(new HierObjectId(ehrId.toString()));
-            objByReference.setEhrStatus(ehrStatus);
-            objByReference.setSystemId(new HierObjectId(systemService.getSystemId()));
-            DvDateTime timeCreated = ehrService.getCreationTime(ehrId);
-            objByReference.setTimeCreated(timeCreated.getValue().toString());
-            // objByReference.setCompositions(null);    // TODO get actual data from service layer
-            // objByReference.setContributions(null);   // TODO get actual data from service layer
-        }
-
-        // create and supplement headers with data depending on which headers are requested
-        HttpHeaders respHeaders = new HttpHeaders();
-        for (String header : headerList) {
-            switch (header) {
-                case CONTENT_TYPE:
-                    if (minimalOrRepresentation != null) {
-                        // if response is going to have a body
-                        respHeaders.setContentType(contentType);
-                    }
-                    break;
-                case LOCATION:
-                    try {
-                        URI url = createLocationUri(EHR, ehrId.toString());
-                        respHeaders.setLocation(url);
-                    } catch (Exception e) {
-                        throw new InternalServerException(e.getMessage());
-                    }
-                    break;
-                case ETAG:
-                    respHeaders.setETag("\"" + ehrId + "\"");
-                    break;
-                case LAST_MODIFIED:
-                    // TODO should be VERSION.commit_audit.time_committed.value which is not implemented yet - mock for
-                    // now
-                    respHeaders.setLastModified(123124442);
-                    break;
-                default:
-                    // Ignore header
-            }
-        }
+        // initialize HTTP 201 Created body builder
+        return ResponseEntity.status(status)
+                .location(uri)
+                .eTag("\"%s\"".formatted(ehrId.toString()))
+                // TODO should be VERSION.commit_audit.time_committed.value which is not implemented yet - mock for now
+                .lastModified(123124442);
```

Comment:
Should this be replaced now?

Reply:
It was there before, I left it there for now